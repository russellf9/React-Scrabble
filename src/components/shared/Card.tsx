import * as React from "react";
import styled from "../style/styled-components";

const Wrapper = styled.div`
 margin: 10px;
 border 1px solid black;
 max-width: 600px;
 min-width: 500px;
 padding: 40px;
 border-radius: 20px;
`;

interface MyProps {}

export const Card: React.FC<MyProps> = (props) => {
  return <Wrapper>{props.children}</Wrapper>;
};
