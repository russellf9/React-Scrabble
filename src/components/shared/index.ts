import styled from 'styled-components';
import { colors, fonts, fontWeights } from '../../themes';

export interface ButtonPrimaryProps {
    background?: string;
    borderColor?: string;
    borderSize?: number;
    borderRadius?: number;
    color?: string;
    marginTop?: number;
    marginRight?: number;
    marginBottom?: number;
    marginLeft?: number;
}

const Button = styled.button<ButtonPrimaryProps>`
    background: ${colors.darkBlue};
    border-radius: 4px;
    border: 1px solid ${colors.darkBlue};
    color: white;
    font-family: ${fonts.hind}
    font-size: 1em;
    font-weight: ${fontWeights.semiBold}
    margin: 0.5em 1em;
    padding 0.5em 1em;
    text-transform: uppercase;
`;

export { Button };
