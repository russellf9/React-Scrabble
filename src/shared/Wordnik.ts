export const WORDNIK_API = 'http://api.wordnik.com/v4/word.json/';
export const KEY = 'a2a73e7b926c924fad7001ca3111acd55af2ffabf50eb4ae5';