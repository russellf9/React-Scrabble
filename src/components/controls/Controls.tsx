import * as React from "react";
import { Button } from "../shared";

export interface ControlsProps {
  clearSearch: Function;
  isLoading: boolean;
}

export const Controls: React.FC<ControlsProps> = ({
  isLoading,
  clearSearch,
}): JSX.Element => (
  <div>
    <Button
      disabled={isLoading}
      onClick={() => clearSearch()}
      style={{ marginLeft: 10 }}
      value="Clear"
    >
      Clear
    </Button>
  </div>
);
