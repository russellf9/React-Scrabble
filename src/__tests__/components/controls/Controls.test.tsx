import * as React from 'react';
import { shallow } from 'enzyme';
import { Controls, ControlsProps } from '../../../components/controls/Controls';
import { Button } from '../../../components/shared';

const initialProps: ControlsProps = {
    clearSearch: (): void => {
       return;
    },
    isLoading: false,
};

describe('Controls', () => {

    it('renders without crashing', () => {
        // given

        // when
        const wrapper = shallow(<Controls {...initialProps}/>);

        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a Button', () => {
        // given
        // when
        const wrapper = shallow(<Controls {...initialProps}/>);

        // then 
        expect(wrapper.contains(<Button/>));
    });

    it('should not be loading', () => {
        // given
        // when
        const wrapper = shallow(<Controls {...initialProps}/>);
        const props = wrapper.instance().props as ControlsProps;
        // then 
        expect(props.isLoading).toBe(false);
    });
});
