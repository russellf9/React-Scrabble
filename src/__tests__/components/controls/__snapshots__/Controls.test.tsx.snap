// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    clearSearch={[Function]}
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <UNDEFINED
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": 10,
          }
        }
        value="Clear"
      >
        Clear
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Clear",
        "disabled": false,
        "onClick": [Function],
        "style": Object {
          "marginLeft": 10,
        },
        "value": "Clear",
      },
      "ref": null,
      "rendered": "Clear",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": e {
          "baseHash": 1014106698,
          "baseStyle": undefined,
          "componentId": "sc-bdfBwQ",
          "isStatic": false,
          "rules": Array [
            "
    background: ",
            "#2993C8",
            ";
    border-radius: 4px;
    border: 1px solid ",
            "#2993C8",
            ";
    color: white;
    font-family: ",
            "hind",
            "
    font-size: 1em;
    font-weight: ",
            "600",
            "
    margin: 0.5em 1em;
    padding 0.5em 1em;
    text-transform: uppercase;
",
          ],
          "staticRulesId": "",
        },
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-bdfBwQ",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <UNDEFINED
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 10,
            }
          }
          value="Clear"
        >
          Clear
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Clear",
          "disabled": false,
          "onClick": [Function],
          "style": Object {
            "marginLeft": 10,
          },
          "value": "Clear",
        },
        "ref": null,
        "rendered": "Clear",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": e {
            "baseHash": 1014106698,
            "baseStyle": undefined,
            "componentId": "sc-bdfBwQ",
            "isStatic": false,
            "rules": Array [
              "
    background: ",
              "#2993C8",
              ";
    border-radius: 4px;
    border: 1px solid ",
              "#2993C8",
              ";
    color: white;
    font-family: ",
              "hind",
              "
    font-size: 1em;
    font-weight: ",
              "600",
              "
    margin: 0.5em 1em;
    padding 0.5em 1em;
    text-transform: uppercase;
",
            ],
            "staticRulesId": "",
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "shouldForwardProp": undefined,
          "styledComponentId": "sc-bdfBwQ",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
